Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,218 @@
+  - Specific version
+  - Stable, Preview or LTS
+  - Scope - location of installation based on permissions
+
+## Goals/Non-Goals
+
+Goals:
+
+- Be discoverable as other built-in cmdlets
+- By default, download and install the latest stable release to the users CurrentUser Scope
+  - CurrentUser - default installation scope
+  - AllUsers - Administrative users may explicit select this scope
+- Allow users to use remoting for install at scale with Invoke-Command
+- Allow advanced users to specify the version of PowerShell
+- Allow advanced users to specify scope (location of installation)
+- Allow advanced users to specify channel Stable/Preview/LTS
+- Users should receive a warning to continue the installation (ConfirmImpact=High)
+
+Non-goals:
+
+- Removal of PowerShell (Remove-PowerShell7)
+- Updating of PowerShell (Update-PowerShell7)
+- Disconnect scenarios - Users that are air-gapped may download the appropriate release from GitHub.
+- Handling of additional MSI switch options - i.e. Enable PSRemoting
+
+## Specification
+
+Cmdlet name: Install-PowerShell7
+
+Alias: None
+
+The default behavior is to download and extract the .ZIP package, then install the latest stable
+release of PowerShell into the user scope allowed by permissions. If with administrative privilege,
+install to the scope of AllUsers. If user does not have administrative privilege, install to the
+scope CurrentUser.
+
+- The package install location - $env:LOCALAPPDATA\Microsoft\PowerShell
+- The installer should update the Users PATH for pwsh
+- The installer should enable Microsoft Updates for scope AllUsers. Due to permission requirements,
+  the scope CurrentUser will require manual updating.
+- Confirm impact should be set to **High** to force a warning and the ability to abort
+
+The following parameters can be added by experienced PowerShell users to customize the installation.
+
+- Parameter: Version - overrides the default and allows the user to specify the version of
+  PowerShell that will be installed. PowerShell version 7.0 and 7.1 are out of support, so version
+  numbers should begin with 7.2. This would install the latest 7.2.x.
+- Parameter: Scope - overrides the default and allows the user to specify the installation location
+  (scope) AllUsers or CurrentUser.
+- Parameter: Channel - options Stable/Preview/LTS - allows the user to select if they want the
+  current preview or Long Term Service (LTS) release. Preview should not be allowed if value of
+  parameter **-Version** is Generally Available (GA).
+- Parameter: Passthru - to provide informative output post installation
+
+## Syntax and Parameter Sets
+
+### ParameterSet Version (Default)
+
+```
+Install-PowerShell7 [[-Version] <string>] [-Scope {AllUsers | CurrentUsers}] [-Passthru] [-WhatIf] [-Confirm]  [<CommonParameters>]
+```
+
+### ParameterSet Channel
+
+```
+Install-PowerShell7 -Channel {Preview | LTS | Stable} [-Scope {AllUsers | CurrentUser}] [-Passthru] [-WhatIf] [-Confirm]
```

Comment:
couldn't there be multiple versions of a channel?

Reply:
For simplification and to remove the need for versions, the installation of a channel should be the latest version at the time of execution.